test/volumes-basic.test.js

```diff
@@ -275,30 +276,20 @@ if (CONFIG.experimental_nfs_shared_volumes !== true) {
             });
     });
 
-    test('volume should eventually transition to state \'deleted\'',
-        function (t) {
-            var expectedState = 'deleted';
-
-            mod_testVolumes.waitForTransitionToState(CLIENT, testVolume.id,
-                expectedState, function onTransition() {
-                    CLIENT.get('/my/volumes/' + testVolume.id,
-                        function onGetVolume(getVolumeErr, req, res, volume) {
-                            t.ifErr(getVolumeErr,
-                                'getting newly created volume should not ' +
-                                    'error');
-                            t.ok(typeof (volume) === 'object' &&
-                                volume !== null,
-                                    'response should be a non-null object');
-                            t.equal(volume.name, testVolumeName,
-                                'volume name should be \'' + testVolumeName +
-                                    '\'');
-                            t.equal(volume.state, expectedState,
-                                'volume should have transitioned to state \'' +
-                                    expectedState + '\'');
-
-                            t.end();
-                    });
-            });
+    test('volume should eventually disappear', function (t) {
+        mod_testVolumes.waitForDeletion(CLIENT, testVolume.id,
+            function onDeleted() {
+                CLIENT.get('/my/volumes/' + testVolume.id,
+                    function onGetVolume(getVolumeErr) {
+                        t.ok(verror.hasCauseWithName(getVolumeErr,
```

Comment:
Doesn't `waitForDeletion` calls its callback with no error if the GetVolume responded with a `VOLUME_NOT_FOUNDError`?